File: back/back/Models/Secteur.cs
using LiteDB;

namespace back.Models;

public sealed class Secteur
{
     [BsonId]
     public int Id { get; set; }

     public string Nom { get; set; } = null!;

     public string CouleurHexa { get; set; } = null!;
}

public sealed class SecteurZoneCarte
{
     [BsonId]
     public int Id { get; set; }

     public Secteur Secteur { get; set; } = null!;

     public int PositionX { get; set; }

     public int PositionY { get; set; }
}

File: front/src/models/Secteur.ts
import { SystemePositionRequete } from "./Systeme"

export type Secteur =
{
    id: number,
    nom: string,
    couleurHexa: string,
    listePosition: SystemePositionRequete[]
}

export type SecteurSynchroniser =
{
    listeCaseAjouter: { positionX: number, positionY: number, idSecteur: number }[],
    listeCaseModifier: { positionX: number, positionY: number, idSecteurNouveau: number }[],
    listeCaseSupprimer: SystemePositionRequete[]
}

export type SecteurRequete =
{
    nom: string,
    couleurHexa: string
}